### PR TITLE
Move all NFD Prow jobs to Community cluster

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
As seen at https://groups.google.com/a/kubernetes.io/g/dev/c/p6PAML90ZOU

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
Fixes https://github.com/kubernetes-sigs/node-feature-discovery/issues/1747

File: config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml [hunk 1]
```diff
@@ -21,7 +22,15 @@ postsubmits:
...
               key: codecov-token
+        resources:
+          limits:
+            cpu: 4
+            memory: 8Gi
+          requests:
+            cpu: 4
+            memory: 8Gi        
   - name: postsubmit-node-feature-discovery-e2e-test
+    cluster: eks-prow-build-cluster
```
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml:33]: if you're doing GCP e2e tests you should use the GCP cluster (k8s-infra-prow-build)

Comment (BenTheElder) [reply config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml:33]: (other jobs like unit test / verify, AWS e2e can and should use the EKS cluster to spread the load between the cloud credits)

**Comment by BenTheElder** [comment]
failing linters

File: config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 postsubmits:
   kubernetes-sigs/node-feature-discovery:
   - name: postsubmit-node-feature-discovery-verify-master
+    cluster: k8s-infra-prow-build
```
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml:4]: this one should probably be on the EKS prow, and it must have been moved there previously as it's using `public.ecr.aws/docker/library/golang:1.22` 

it's not doing e2e is it?

Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml:4]: https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/scripts/test-infra/verify.sh 
so yeah, not e2e

Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml:4]: should I move it to EKS?

Comment (BenTheElder) [reply config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml:4]: yes

File: config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml [hunk 3]
```diff
@@ -59,3 +68,10 @@ postsubmits:
...
+        resources:
```
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml:71]: TODO: evaluate the necessary amount for this job (happy to defer that and get it migrated first)

File: config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml (hunk 2 above)
Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml]: done

File: config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml (hunk 3 above)
Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml:71]: for our e2e we have tried 2 cpu's but it runs out of time. 4C/8M has been the min set , but happy to evaluate if we can trim down a bit

**Comment by ArangoGutierrez** [comment]
> failing linters

the lints that trigger the failure are not on NFD files, how can we pass by this, should I rebase on master or something?

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve

**Comment by BenTheElder** [comment]
```
config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-postsubmits.yaml
  31:24     error    trailing spaces  (trailing-spaces)
  77:26     error    no new line character at the end of file  (new-line-at-end-of-file)
  77:24     error    trailing spaces  (trailing-spaces)
```
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/32887/pull-test-infra-verify-lint/1808204092611235840

**Comment by BenTheElder** [comment]
> the lints that trigger the failure are not on NFD files, how can we pass by this, should I rebase on master or something?

the other lints are warnings, which are not the failure, NFD is failing (see above, you can click to expand the log lines for more detail)

**Comment by ArangoGutierrez** [comment]
> > the lints that trigger the failure are not on NFD files, how can we pass by this, should I rebase on master or something?
> 
> the other lints are warnings, which are not the failure, NFD is failing (see above, you can click to expand the log lines for more detail)

Fixed

**Review by BenTheElder — COMMENTED** [review]
/lgtm

**Pull request merged**